lib/modules/profiler/index.js

```diff
@@ -1,28 +1,34 @@
 const asciiTable = require('ascii-table');
+const GasEstimator = require('../gasEstimator');
 
 class Profiler {
   constructor(embark) {
     this.embark = embark;
     this.logger = embark.logger;
     this.events = embark.events;
+    this.gasEstimator = new GasEstimator(embark);
 
     this.registerConsoleCommand();
   }
 
   profile(contractName, contract) {
     const self = this;
     let table = new asciiTable(contractName);
-    table.setHeading('Function', 'Payable', 'Mutability', 'Inputs', 'Outputs');
-    contract.abiDefinition.forEach((abiMethod) => {
-      switch(abiMethod.type) {
-        case "constructor": 
-          table.addRow("constructor", abiMethod.payable, abiMethod.stateMutability, this.formatParams(abiMethod.inputs), this.formatParams(abiMethod.outputs));
-          break;
-        default:
-          table.addRow(abiMethod.name, abiMethod.payable, abiMethod.stateMutability, this.formatParams(abiMethod.inputs), this.formatParams(abiMethod.outputs));
-      }
+    table.setHeading('Function', 'Payable', 'Mutability', 'Inputs', 'Outputs', 'Gas Estimates');
+    self.gasEstimator.estimateGas(contractName, function(err, gastimates) {
+      if (err) throw new Error(err);
```

Comment:
Should probably not throw as that would stop the dashboard and users might not see the error. Probably better to log it and just stop the profiling.